tests/Doctrine/Tests/DBAL/Schema/IndexTest.php

```diff
@@ -188,4 +189,40 @@ public function testOptions() : void
         self::assertSame('name IS NULL', $idx2->getOption('WHERE'));
         self::assertSame(['where' => 'name IS NULL'], $idx2->getOptions());
     }
+
+    /**
+     * @param int[] $lengths1
+     * @param int[] $lengths2
+     *
+     * @dataProvider spatialIndexProvider
+     */
+    public function testFulfilledSpatialIndex(?string $flag1, ?string $flag2, array $lengths1, array $lengths2, bool $expected) : void
+    {
+        $index1 = new Index('index1', ['geo_location'], false, false, [], ['lengths' => $lengths1]);
+        if ($flag1 !== null) {
+            $index1->addFlag($flag1);
+        }
+
+        $index2 = new Index('index2', ['geo_location'], false, false, [], ['lengths' => $lengths2]);
+        if ($flag2 !== null) {
+            $index2->addFlag($flag2);
+        }
+
+        self::assertSame($expected, $index1->isFullfilledBy($index2));
+        self::assertSame($expected, $index2->isFullfilledBy($index1));
+    }
+
+    public static function spatialIndexProvider() : Generator
+    {
+        yield ['spatial', 'spatial', [], [], true];
+        yield ['spatial', 'spatial', [], [32], true];
+        yield ['spatial', 'spatial', [32], [32], true];
+        yield ['spatial', 'spatial', [32], [24], true];
```

Comment:
Since this is a data provider for a spatial index test, why do we care about testing the length comparison? If the logic of comparison of the flags _and_ lengths needs to be tested, it's better to do it in a separate test that would not get too much into the details of why the lengths or flags mismatch.

Reply:
We are testing if the method `isFullfilledBy` works as expected in different cases. When spatial index created or migration is generated (or checked) on different platforms: MySQL, AWS Aurora MySQL.

Reply:
This is fine but some of these different cases are better to be implemented as a separate test method. Otherwise, if someone breaks the length comparison logic and `testFulfilledSpatialIndex()` fails, it will be misleading feedback from the test suite.

Reply:
Using a text key on data providers would also give more information to a failing test with a data row, but separate test cases provide a more contextual structure for the test class.